semantics/language/common/typing/compatibility.k

```diff
@@ -352,8 +352,10 @@ module C-TYPING-COMPATIBILITY
           => t(Mods, functionType(T, P'))
      rule compositeFunctionType(T:Type, Mods:Set, .K, .List, P:List, _)
```

Comment:
I think the way this was is left over from a limitation of the Java rewrite engine which is no longer present, wherein such rewrites in function rules were not allowed. In any event, I'm not sure it's worth the effort to change at this point.